canonical-websites/docs.ubuntu.com
# Fix error pages and redirects

This does a bunch of things to fix #208 and some Django v2 upgrade oversights:

- Redirect `/landscape/en/onprem-overview` to `/landscape/en/onprem`
- Remove broken custom logic for error pages
- Tidy up templates for 404 and 500 pages a bit
- Upgrade whitenoise to v4
- Replace unslashed with removing slash in `redirects.yaml`
- Set `DJANGO_DEBUG` properly

QA
--

`./run clean && ./run` (this might take a little while)

- Go to http://0.0.0.0:8007/landscape/en/onprem-overview, check you're redirected to http://0.0.0.0:8007/landscape/en/onprem.
- Go to http://0.0.0.0:8007/landscape/en/onprem/ or http://0.0.0.0:8007/landscape/en/onprem.html, check you're redirected to http://0.0.0.0:8007/landscape/en/onprem
- Go to http://0.0.0.0:8007/landscape/en/onprem-erroneous, check you see a Django debug page about a 404

Now kill the server. Open up `webapp/urls.py` and add an "error" path lines:

``` python3
from django.http import HttpResponseServerError
...
urlpatterns += [
    url("error", lambda request: HttpResponseServerError()),
    url(r"^(?P<template>.*)/?$", TemplateFinder.as_view())
]
```

Now run `./run --env DJANGO_DEBUG=false`, and:

- Go to http://0.0.0.0:8007/landscape/en/onprem-erroneous - check you see a relatively nice 404 page
- Go to http://0.0.0.0:8007/error - check you see a relatively nice 500 server error page
- Browse the site generally, check everything works (e.g. http://0.0.0.0:8007/, http://0.0.0.0:8007/landscape, http://0.0.0.0:8007/landscape/en)
